api/raft.cc

```diff
@@ -83,12 +83,40 @@ void set_raft(http_context&, httpd::routes& r, sharded<service::raft_group_regis
             return json_return_type(srv.current_leader().to_sstring());
         });
     });
+    r::read_barrier.set(r, [&raft_gr] (std::unique_ptr<http::request> req) -> future<json_return_type> {
+        auto timeout = get_request_timeout(*req);
+
+        if (!req->query_parameters.contains("group_id")) {
+            // Read barrier on group 0 by default
+            co_await raft_gr.invoke_on(0, [timeout] (service::raft_group_registry& raft_gr) -> future<> {
+                co_await raft_gr.group0_with_timeouts().read_barrier(nullptr, timeout);
```

Comment:
nit: pessimizing coroutine (by using `return` here instead of `co_await`, we'd avoid one allocation, without losing any readability or anything).

Doesn't matter though, this is not a hot path, so we can keep it like that. (But it would potentially be important for hot path code -- good to remember in the future.)